src/Framework/PaymentGateways/PaymentGateway.php

```diff
@@ -148,30 +185,118 @@ public function cancelSubscription(Subscription $subscription)
     }
 
     /**
+     * @unreleased
      * @inheritDoc
+     * @throws Exception
      */
     public function canSyncSubscriptionWithPaymentGateway(): bool
     {
-        return $this instanceof SubscriptionTransactionsSynchronizable
-            || $this->subscriptionModule->canSyncSubscriptionWithPaymentGateway();
+        if ($this->subscriptionModule) {
+            return $this->subscriptionModule->canSyncSubscriptionWithPaymentGateway();
+        }
+
+        throw new Exception(esc_html__('Method has not been implemented yet.', 'give'));
     }
 
     /**
+     * @unreleased
      * @inheritDoc
+     * @throws Exception
      */
     public function canUpdateSubscriptionAmount(): bool
     {
-        return $this instanceof SubscriptionAmountEditable
-            || $this->subscriptionModule->canUpdateSubscriptionAmount();
+        if ($this->subscriptionModule) {
+            return $this->subscriptionModule->canUpdateSubscriptionAmount();
+        }
+
+        throw new Exception(esc_html__('Method has not been implemented yet.', 'give'));
```

Comment:
As is elsewhere, if we can do the check by seeing if `$this` is using an interface, we should. Runtime "not implemented" exceptions do have their place but should not to be reached for too soon. If they choose not to override and aren't using the interface then this can safely return false.

Reply:
replied in another thread.